src/convertToObject.js

```diff
@@ -3,10 +3,32 @@
 /**
  * @param {string} sourceString
  *
- * @return {object}
+ * @return {object}n
  */
-function convertToObject(sourceString) {
-  // write your code here
-}
+function convertToObject(stylesString) {
+  // Создаем копию параметра, чтобы не изменять его
+  const modifiedStylesString = stylesString
+    .replace(/;+\s*$/, '') // Удаляет точки с запятой в конце строки
+    .trim();
+
+  const stylesObject = {};
+
+  // Разделяем строки на отдельные правила
+  const rules = modifiedStylesString.split(/;\s*/);
```

Comment:
Instead of using `split` and then `forEach` to iterate over the rules, consider using `split` followed by `filter` to remove any empty rules and then `map` to transform the remaining rules into the desired object structure. This would adhere to the checklist item about using iteration methods and avoiding loops.